Update go version to 1.23.6

AnnaZivkovic opened the pull request:
Image: registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.23-openshift-4.19
Go version: 1.23.6
```
oc image info registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.23-openshift-4.19
error: the image is a manifest list and contains multiple images - use --filter-by-os to select from:

  OS            DIGEST
  linux/amd64   sha256:577fdb5531667fd5387702469e81bd12507508dcf2dad37f056970e0974c81ac
  linux/arm64   sha256:ad9094bf51086a50bbe3dc9915f8e62f0526051e4fbe2011f334692af290ccc0
  linux/ppc64le sha256:ecf277029dd79dbfce3ec1e10db8df5cf775e1270df394fc470dcab1e005bf48
  linux/s390x   sha256:07943a16db3053d14c001a49ca729fbfce99b1548a5c2d5ff97d4eeea7ccfe61
```
Using Kubernetes version 1.32.3 [based on](https://mirror.openshift.com/pub/openshift-v4/amd64/clients/ocp-dev-preview/4.19.0-ec.4/release.txt)  

```
Name:           4.19.0-ec.4
Digest:         sha256:1e9baecce90105b4c0aa695dadc2fac7e4b9fd54e0d2ede90e143e57df8e424a
Created:        2025-03-31T07:58:36Z
OS/Arch:        linux/amd64
Manifests:      772
Metadata files: 2
...

Component Versions:
  kubectl          1.32.1                
  kubernetes       1.32.3                
  kubernetes-tests 1.32.2                
  machine-os       419.96.202503251418-0 Red Hat Enterprise Linux CoreOS
```
Steps taken in each commit
1. Update the Dockerfiles to use a base image with the desired Golang 
    Update the Makefile to use the new Golang version base image (BUILD_IMAGE variable)
2. Update the k8s libraries in the go.mod file to the desired version
     Used `go get -u`, and ensured correct go version was used
3. `make vendor`
4. updated api calls to accommodate newer versions 
     - `github.com/openshift/library-go/pkg/operator/events`

5. Update the tools in the Makefile to the desired version:

```makefile
# https://github.com/kubernetes-sigs/kustomize/releases
KUSTOMIZE_VERSION ?= v5.6.0
# https://github.com/kubernetes-sigs/controller-tools/releases
CONTROLLER_TOOLS_VERSION ?= v0.17.2
# https://github.com/kubernetes-sigs/controller-runtime/branches
SETUP_ENVTEST_VERSION ?= release-0.20
# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
ENVTEST_K8S_VERSION = 1.32.1
# https://github.com/golangci/golangci-lint/releases
GOLINT_VERSION = v2.0.2
```
6. Ran `make manifest`, `make generate`, `make bundle`
7. Fix staticchekc warnings that arose with newer versions

AnnaZivkovic commented:
/retest

AnnaZivkovic commented on .ci-operator.yaml:
```diff
@@ -1,4 +1,4 @@
   name: builder
   namespace: ocp
-  tag: rhel-9-golang-1.22-builder-multi-openshift-4.17
+  tag: rhel-9-golang-1.23-openshift-4.19
```
```
oc image info registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.23-openshift-4.19
error: the image is a manifest list and contains multiple images - use --filter-by-os to select from:

  OS            DIGEST
  linux/amd64   sha256:577fdb5531667fd5387702469e81bd12507508dcf2dad37f056970e0974c81ac
  linux/arm64   sha256:ad9094bf51086a50bbe3dc9915f8e62f0526051e4fbe2011f334692af290ccc0
  linux/ppc64le sha256:ecf277029dd79dbfce3ec1e10db8df5cf775e1270df394fc470dcab1e005bf48
  linux/s390x   sha256:07943a16db3053d14c001a49ca729fbfce99b1548a5c2d5ff97d4eeea7ccfe61
```

Prashanth684 commented:
/test unit

Prashanth684 commented on Makefile:
```diff
@@ -71,15 +71,15 @@ IMG ?= registry.ci.openshift.org/origin/multiarch-tuning-operator:main
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
-ENVTEST_K8S_VERSION = 1.29.3
+ENVTEST_K8S_VERSION = 1.32.1
 # https://github.com/golangci/golangci-lint/releases
```
are these the correct updates? are we missing something. the unit test is failing with:
```
2025-04-18T05:11:58Z	ERROR	controller-runtime.test-env	unable to start the controlplane	{"tries": 0, "error": "exec: \"etcd\": executable file not found in $PATH"}
  sigs.k8s.io/controller-runtime/pkg/envtest.(*Environment).startControlPlane
  	/go/src/github.com/openshift/multiarch-tuning-operator/vendor/sigs.k8s.io/controller-runtime/pkg/envtest/server.go:328
  sigs.k8s.io/controller-runtime/pkg/envtest.(*Environment).Start
  	/go/src/github.com/openshift/multiarch-tuning-operator/vendor/sigs.k8s.io/controller-runtime/pkg/envtest/server.go:249
```

Prashanth684 commented:
is it possible to split these into a separate PR for the golang update and one for the k8s update? that way we can isolate the issues and get things merged faster.

aleskandro commented:
I would be ok with splitting too. Especially as we approach the z-stream release for the cve fixes. Not strongly opinioned about it though. I imagine that the CI here will fail until https://github.com/openshift/release/pull/63835 merges.

aleskandro commented:
If we want to split, we can merge the libs updates before https://github.com/openshift/release/pull/63835, then https://github.com/openshift/release/pull/63835 and the go update.

aleskandro commented:
/cherry-pick v1.0

AnnaZivkovic commented:
Just rebased and https://github.com/openshift/release/pull/63835 is merged. Going to see how the tests do. If there are a lot of problems Ill split it out into two prs. Id like to avoid doing that as it is a lot work.

AnnaZivkovic commented:
/retest

AnnaZivkovic commented:
/retest

AnnaZivkovic commented:
We can not split this pr into two as the updates are codependent

aleskandro commented:
> is it possible to split these into a separate PR for the golang update and one for the k8s update? that way we can isolate the issues and get things merged faster.

Some dependencies strongly require the new version of golang (k8s libs in particular).

aleskandro commented on hack/golangci-lint.sh:
```diff
@@ -3,4 +3,4 @@
 echo "Running golangci-lint..."
 GOFLAGS='' go install "github.com/golangci/golangci-lint/v2/cmd/golangci-lint@${GOLINT_VERSION:-latest}"
-golangci-lint run --timeout 5m0s --verbose --skip-dirs vendor --skip-files zz_generated*
+golangci-lint run --verbose
```
what about the other arguments?

aleskandro commented on pkg/testing/registry/registry.go:
```diff
@@ -18,7 +18,6 @@ import (
 
 	"github.com/google/uuid"
-	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
```
can we rather use corev1 here to give explicit info about what `v1` we are using there?

aleskandro commented on pkg/testing/framework/utils.go:
```diff
@@ -14,7 +14,6 @@ import (
 
 	admissionv1 "k8s.io/api/admissionregistration/v1"
 	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
```
same as in https://github.com/openshift/multiarch-tuning-operator/pull/542/commits/6f212eec27aaa3a40e1cc057673a6190cd82b8cd#r2068131346

aleskandro commented on .golangci.bck.yaml:
```diff
@@ -0,0 +1,102 @@
+linters:
```
can we remove this? we can restore by looking back in the git history

aleskandro commented:
/cherry-pick v1.x

aleskandro commented:
/override ?

aleskandro commented:
/override "Red Hat Konflux / run-prowjob-bundle / multiarch-tuning-operator-bundle" "Red Hat Konflux / run-prowjob / multiarch-tuning-operator"

aleskandro commented:
/override pull-ci-openshift-multiarch-tuning-operator-main-ocp419-e2e-gcp
(known bug)

aleskandro commented:
/lgtm
/approve

aleskandro commented:
/override ?

aleskandro commented:
/override "Red Hat Konflux / run-prowjob-bundle / multiarch-tuning-operator-bundle" "Red Hat Konflux / run-prowjob / multiarch-tuning-operator"

aleskandro commented:
/override pull-ci-openshift-multiarch-tuning-operator-main-ocp419-e2e-gcp

/retest

aleskandro commented:
/override ?

aleskandro commented:
/override "Red Hat Konflux / run-prowjob-bundle / multiarch-tuning-operator-bundle" "Red Hat Konflux / run-prowjob / multiarch-tuning-operator"

aleskandro commented:
/override "Red Hat Konflux / run-prowjob-bundle / multiarch-tuning-operator-bundle" "Red Hat Konflux / run-prowjob / multiarch-tuning-operator"

AnnaZivkovic commented:
List of errors that occurred  from updating staticcheck

```
          "User": "",
          "History": [
               {
                    "created": "2024-09-18T12:17:53.005564697Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:d68d95268102e3ea8d2fd179a3b78a7a0ec5a64a6870605e5946920f12db94e8 in / ",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:53.584837972Z",
                    "created_by": "/bin/sh -c mv -f /etc/yum.repos.d/ubi.repo /tmp || :",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:53.802667604Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:5b1f650e1376d79fa3a65df4a154ea5166def95154b52c1c1097dfd8fc7d58eb in /tmp/tls-ca-bundle.pem ",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075372561Z",
                    "created_by": "/bin/sh -c #(nop) ADD multi:5ae7ffc56bcb5cc7fa59a3ed8016921afd11d799db8be1738c243f37637c35c7 in /etc/yum.repos.d/ ",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.07541457Z",
                    "created_by": "/bin/sh -c #(nop) LABEL maintainer=\"Red Hat, Inc.\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075505853Z",
                    "created_by": "/bin/sh -c #(nop) LABEL com.redhat.component=\"rhel-els-container\"       name=\"rhel-els\"       version=\"9.4\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075537612Z",
                    "created_by": "/bin/sh -c #(nop) LABEL com.redhat.license_terms=\"https://www.redhat.com/en/about/red-hat-end-user-license-agreements#UBI\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075563592Z",
                    "created_by": "/bin/sh -c #(nop) LABEL summary=\"Provides the latest release of the Red Hat Extended Life Base Image.\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075701891Z",
                    "created_by": "/bin/sh -c #(nop) LABEL description=\"The Extended Life Base Image is designed and engineered to be the base layer for Red Hat extended life products. This base image is only supported for approved Red Hat products. This image is maintained by Red Hat and updated regularly.\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075738017Z",
                    "created_by": "/bin/sh -c #(nop) LABEL io.k8s.display-name=\"Red Hat Extended Life Base Image\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.07575268Z",
                    "created_by": "/bin/sh -c #(nop) LABEL io.openshift.expose-services=\"\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075772617Z",
                    "created_by": "/bin/sh -c #(nop) LABEL io.openshift.tags=\"base rhel9\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075782464Z",
                    "created_by": "/bin/sh -c #(nop) ENV container oci",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075859912Z",
                    "created_by": "/bin/sh -c #(nop) ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.075880273Z",
                    "created_by": "/bin/sh -c #(nop) CMD [\"/bin/bash\"]",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:54.684694513Z",
                    "created_by": "/bin/sh -c rm -rf /var/log/*",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:55.282901214Z",
                    "created_by": "/bin/sh -c mkdir -p /var/log/rhsm",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:55.509463182Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:b4285c27fba0283f525ca705599d2f9a83ce5fd4a1aa10f7cf92cc0254bbecdc in /root/buildinfo/content_manifests/rhel-els-container-9.4-943.1726661131.json ",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:55.758417588Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:c04f3aad027364072a21769f30429c942f68f7a99f301f8bb8a1f8f42ac685ec in /root/buildinfo/Dockerfile-rhel-els-9.4-943.1726661131 ",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:55.758659544Z",
                    "created_by": "/bin/sh -c #(nop) LABEL \"release\"=\"943.1726661131\" \"distribution-scope\"=\"public\" \"vendor\"=\"Red Hat, Inc.\" \"build-date\"=\"2024-09-18T12:09:15\" \"architecture\"=\"x86_64\" \"vcs-type\"=\"git\" \"vcs-ref\"=\"929523a15010ff907ed9a3af3ead8c8d0c5ca350\" \"io.k8s.description\"=\"The Extended Life Base Image is designed and engineered to be the base layer for Red Hat extended life products. This base image is only supported for approved Red Hat products. This image is maintained by Red Hat and updated regularly.\" \"url\"=\"https://access.redhat.com/containers/#/registry.access.redhat.com/rhel-els/images/9.4-943.1726661131\"",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:56.385153707Z",
                    "created_by": "/bin/sh -c rm -f '/etc/yum.repos.d/odcs-3495402-1c831.repo' '/etc/yum.repos.d/rhel-9.4-compose-34ae9.repo'",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:57.268296574Z",
                    "created_by": "/bin/sh -c rm -f /tmp/tls-ca-bundle.pem",
                    "empty_layer": true
               },
               {
                    "created": "2024-09-18T12:17:59.74073854Z",
                    "created_by": "/bin/sh -c mv -fZ /tmp/ubi.repo /etc/yum.repos.d/ubi.repo || :"
               },
               {
                    "created": "2025-03-04T14:56:59.754733233Z",
                    "created_by": "/bin/sh -c mv -f /etc/yum.repos.d/ubi.repo /tmp || :",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:56:59.934629739Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:df13b6a4fdeac4626ab7cd7e99d16d30edc6e3e5441295ae7691bee26604a5b1 in /tmp/tls-ca-bundle.pem ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:57:00.119542868Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:b9e62db45a66434a3642ca060fe7dd24393453808674c0d04dcdead0bf5e04cd in /etc/yum.repos.d/ ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:57:00.119638669Z",
                    "created_by": "/bin/sh -c #(nop) ENV __doozer=update __doozer_group=rhel-9-golang-1.23 __doozer_key=openshift-golang-builder __doozer_version=v1.23.6 ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:57:00.119729626Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_BUILD_ENGINE=brew",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:57:00.119749015Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_BUILD_DEPS_METHOD=cachito",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:57:00.119779609Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_BUILD_NETWORK=internal-only",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:57:00.11979534Z",
                    "created_by": "/bin/sh -c #(nop) ARG ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:57:00.120029066Z",
                    "created_by": "/bin/sh -c #(nop) ENV SUMMARY=\"RHEL9 based Go builder image for OpenShift ART\"     container=oci     GOFLAGS='-mod=vendor'     GOPATH=${GOPATH:-/go}     GOMAXPROCS=8     GOAMD64=v2     VERSION=\"1.23\"     GO_VERSION=\"${__doozer_version:-$VERSION}\"     GODEBUG=\"disablethp=1\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:59:26.780117308Z",
                    "created_by": "/bin/sh -c dnf update -y &&     dnf install -y --nodocs         bc         diffutils         dos2unix         file         findutils         git         goversioninfo         gpgme         gpgme-devel         hostname         krb5-devel         libassuan-devel         libtool         lsof         make         openssl         openssl-devel         patch         python3         rsync         socat         systemd-devel         tar         tree         util-linux         wget         which         xz         zip &&     dnf install -y \"golang-*$VERSION*\" &&     mkdir -p /go/src",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T14:59:28.944270267Z",
                    "created_by": "/bin/sh -c #(nop) COPY file:a95817ada61675339ba4962f7bfc8d0c1d862bce71e3464c8d2de3bc2630ef37 in . ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:55.224572364Z",
                    "created_by": "/bin/sh -c [ $(go env GOARCH) != \"amd64\" ] || (    yum install -y --setopt=tsflags=nodocs     llvm-toolset cmake3 gcc-c++ libxml2-devel     glibc mingw64-gcc &&     tar zfx cross.tar.gz &&     export TP_OSXCROSS_DEV=$(pwd)/cross/deps &&     pushd cross/osxcross &&     UNATTENDED=yes ./build.sh &&     popd &&     cp -avr cross/osxcross/target/bin/* /usr/local/bin/ &&     cp -avr cross/osxcross/target/lib/* /usr/local/lib64/ &&     cp -avr cross/osxcross/target/SDK /usr/local/SDK &&     echo /usr/local/lib64 > /etc/ld.so.conf.d/local.conf &&     /sbin/ldconfig &&     rm -rf cross)",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:56.321754872Z",
                    "created_by": "/bin/sh -c rm -f cross.tar.gz && yum clean all -y",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:56.510598858Z",
                    "created_by": "/bin/sh -c #(nop) COPY file:c50fdddef44667efd8faf7821f8379e83e1aaf2d49ca2e73ebbae09edda1f9f2 in /tmp/go_wrapper.sh ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:57.082832116Z",
                    "created_by": "/bin/sh -c GO_BIN_PATH=$(which go) && mv $GO_BIN_PATH $GO_BIN_PATH.real && mv /tmp/go_wrapper.sh $GO_BIN_PATH && chmod +x $GO_BIN_PATH",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:57.083069363Z",
                    "created_by": "/bin/sh -c #(nop) LABEL         summary=\"RHEL9 based Go builder image for OpenShift ART\"         description=\"RHEL9 based Go builder image for OpenShift ART\"         io.k8s.description=\"golang builder image for Red Hat internal builds\"         io.k8s.display-name=\"Go Builder 1.23\"         com.redhat.license_terms=\"https://www.redhat.com/en/about/red-hat-end-user-license-agreements#UBI\"         version=\"v1.23.6\"         name=\"openshift/golang-builder\"         com.redhat.component=\"openshift-golang-builder-container\"         io.openshift.maintainer.project=\"OCPBUGS\"         io.openshift.maintainer.component=\"Security\"         release=\"202503041452.g6c23478.el9\"         io.openshift.build.commit.id=\"6c234787d92507d06bd80d5c54b022462aa75978\"         io.openshift.build.source-location=\"https://github.com/openshift-eng/ocp-build-data\"         io.openshift.build.commit.url=\"https://github.com/openshift-eng/ocp-build-data/commit/6c234787d92507d06bd80d5c54b022462aa75978\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:57.279047748Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:aa9b60952481337dc15963834f1d0c381cc9f813137d7a8838eb2181e6e92731 in /root/buildinfo/content_manifests/openshift-golang-builder-container-v1.23.6-202503041452.g6c23478.el9.json ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:57.467516865Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:067b3b16f8a998e947005efeb02416e72dc87f33106f3aaee49e8ecbaef27796 in /root/buildinfo/Dockerfile-openshift-golang-builder-v1.23.6-202503041452.g6c23478.el9 ",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:57.467624506Z",
                    "created_by": "/bin/sh -c #(nop) LABEL \"distribution-scope\"=\"public\" \"vendor\"=\"Red Hat, Inc.\" \"build-date\"=\"2025-03-04T14:54:20\" \"architecture\"=\"x86_64\" \"vcs-type\"=\"git\" \"vcs-ref\"=\"7a3542b9a5e54ca58f9d00148022dba3ad79a705\" \"url\"=\"https://access.redhat.com/containers/#/registry.access.redhat.com/openshift/golang-builder/images/v1.23.6-202503041452.g6c23478.el9\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:58.037580544Z",
                    "created_by": "/bin/sh -c rm -f '/etc/yum.repos.d/unsigned-b7e6b.repo'",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:03:58.603289735Z",
                    "created_by": "/bin/sh -c rm -f /tmp/tls-ca-bundle.pem",
                    "empty_layer": true
               },
               {
                    "created": "2025-03-04T15:04:48.981747413Z",
                    "created_by": "/bin/sh -c mv -fZ /tmp/ubi.repo /etc/yum.repos.d/ubi.repo || :",
                    "comment": "FROM registry-proxy.engineering.redhat.com/rh-osbs/rhel-els@sha256:2aaaf576ca73226a6f00af0fd0f0f1e08d5f6a269d15f9b5305d3d5bde94d4f7"
               },
               {
                    "created": "2025-04-16T18:43:34.501149267Z",
                    "created_by": "/bin/sh -c mv -f /etc/yum.repos.d/ubi.repo /tmp || :",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.682255052Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:df13b6a4fdeac4626ab7cd7e99d16d30edc6e3e5441295ae7691bee26604a5b1 in /tmp/tls-ca-bundle.pem ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865186834Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:dfcb1dbb34a6733954f99d44448bbbb136b974ba18d6577b1ef9820752e4d0d8 in /etc/yum.repos.d/ ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865291458Z",
                    "created_by": "/bin/sh -c #(nop) ENV __doozer=update __doozer_group=openshift-4.19 __doozer_key=ci-openshift-golang-builder-latest.rhel9 __doozer_version=v4.19.0 ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865362972Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_BUILD_ENGINE=brew",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865389452Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_BUILD_DEPS_METHOD=cachito",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.86542531Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_BUILD_NETWORK=internal-only",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865459667Z",
                    "created_by": "/bin/sh -c #(nop) ENV OPENSHIFT_CI=\"true\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865500933Z",
                    "created_by": "/bin/sh -c #(nop) ENV GO_COMPLIANCE_INFO=0",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865602006Z",
                    "created_by": "/bin/sh -c #(nop) ENV GOARM=5     LOGNAME=deadbeef     GOCACHE=/go/.cache     GOPATH=/go     LOGNAME=deadbeef",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:34.865656932Z",
                    "created_by": "/bin/sh -c #(nop) ENV PATH=$PATH:$GOPATH/bin",
                    "empty_layer": true
               },
               {
                    "created": "2
                    "created": "2025-04-16T18:43:38.912445293Z",
                    "created_by": "/bin/sh -c chmod 777 /etc/yum.repos.d/",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:38.912527121Z",
                    "created_by": "/bin/sh -c #(nop) ENV DNF_WRAPPER_DIR=/bin/dnf_wrapper",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:38.91265554Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_REPOS_DIR_CI=\"/etc/yum.repos.art/ci\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:38.912708633Z",
                    "created_by": "/bin/sh -c #(nop) ENV ART_REPOS_DIR_LOCALDEV=\"/etc/yum.repos.art/localdev\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:39.092816461Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:f0c34dc715602ce0661b22612d46d0c94b0217e46ccf16f4c3e5eee728849cbb in /tmp ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:39.273715007Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:8e088ba1004eb511ba2cf0e7e6bcf81410dad24a53514c9c5f885a8371ebeba6 in /tmp ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:40.663346504Z",
                    "created_by": "/bin/sh -c chmod +x /tmp/*.sh &&     /tmp/install_dnf_wrapper.sh",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:40.663565325Z",
                    "created_by": "/bin/sh -c #(nop) ENV PATH=$DNF_WRAPPER_DIR:$PATH",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:40.844950888Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:9a0843d1d2b3ef394bd5f64b0fc57c3061dd1054fbdc55e4c86a89d7d7bb16e3 in $DNF_WRAPPER_DIR/ ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:40.845035629Z",
                    "created_by": "/bin/sh -c #(nop) ENV CI_RPM_SVC=base-4-19-rhel9.ocp.svc",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:40.845204043Z",
                    "created_by": "/bin/sh -c #(nop) LABEL         io.k8s.description=\"golang 1.23 builder image for Red Hat CI\"         name=\"openshift/ci-openshift-golang-builder-latest-rhel9\"         com.redhat.component=\"ci-openshift-golang-builder-latest-container\"         io.openshift.maintainer.project=\"OCPBUGS\"         io.openshift.maintainer.component=\"Release\"         release=\"202504161817.p0.g985b057.assembly.stream.el9\"         io.openshift.build.commit.id=\"985b0577010f80824d9371d3dd8dffa233d61c0a\"         io.openshift.build.source-location=\"https://github.com/openshift-eng/ocp-build-data\"         io.openshift.build.commit.url=\"https://github.com/openshift-eng/ocp-build-data/commit/985b0577010f80824d9371d3dd8dffa233d61c0a\"         version=\"v4.19.0\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:41.028818041Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:0e593c91d02af0044d81dc9373b234bed22fa0dd74e22b1cdfaf60d84df9a525 in /root/buildinfo/content_manifests/ci-openshift-golang-builder-latest-container-v4.19.0-202504161817.p0.g985b057.assembly.stream.el9.json ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:41.209714902Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:c762e9b3e5c9db4f475d7e3557dce64846956adf867764eeb5272d63bc51270e in /root/buildinfo/Dockerfile-openshift-ci-openshift-golang-builder-latest-rhel9-v4.19.0-202504161817.p0.g985b057.assembly.stream.el9 ",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:41.209890869Z",
                    "created_by": "/bin/sh -c #(nop) LABEL \"com.redhat.license_terms\"=\"https://www.redhat.com/agreements\" \"distribution-scope\"=\"public\" \"vendor\"=\"Red Hat, Inc.\" \"build-date\"=\"2025-04-16T18:40:58\" \"architecture\"=\"x86_64\" \"vcs-type\"=\"git\" \"vcs-ref\"=\"ecf5e6d2f39024a3562f6ad626838d6b773e926c\" \"description\"=\"golang 1.23 builder image for Red Hat CI\" \"url\"=\"https://access.redhat.com/containers/#/registry.access.redhat.com/openshift/ci-openshift-golang-builder-latest-rhel9/images/v4.19.0-202504161817.p0.g985b057.assembly.stream.el9\"",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:41.71818581Z",
                    "created_by": "/bin/sh -c rm -f '/etc/yum.repos.d/unsigned-64bee.repo'",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:42.243909888Z",
                    "created_by": "/bin/sh -c rm -f /tmp/tls-ca-bundle.pem",
                    "empty_layer": true
               },
               {
                    "created": "2025-04-16T18:43:48.226441542Z",
                    "created_by": "/bin/sh -c mv -fZ /tmp/ubi.repo /etc/yum.repos.d/ubi.repo || :",
                    "comment": "FROM registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder@sha256:4805e1cb2d1bd9d3c5de5d6986056bbda94ca7b01642f721d83d26579d333c60"
               }
          ],
```
